lightning/src/ln/onion_route_tests.rs

```diff
@@ -371,15 +371,15 @@ fn test_onion_failure() {
 		// and tamper returning error message
 		let session_priv = SecretKey::from_slice(&[3; 32]).unwrap();
 		let onion_keys = onion_utils::construct_onion_keys(&Secp256k1::new(), &route.paths[0], &session_priv).unwrap();
-		msg.reason = onion_utils::build_first_hop_failure_packet(&onion_keys[0].shared_secret[..], NODE|2, &[0;0]);
+		msg.reason = onion_utils::build_first_hop_failure_packet(&onion_keys[0].shared_secret.secret_bytes(), NODE|2, &[0;0]);
```

Comment:
Here and elsewhere, should we use `shared_secret.as_ref()` instead of copying the bytes?

Reply:
fixing